### ISSUE TITLE
Subtitles

### ISSUE THREAD

**Issue by asuragan** [rehmatworks/edx-downloader#20]
I'm not a programmer, but I changed the code a little bit to be able to download subtitles from edx

in `def get_course_data(self, course_url):`
![image_2021-06-28_01-14-30](https://user-images.githubusercontent.com/55428472/123561108-49108480-d7af-11eb-91a9-752c9a0797ee.png)

` if 'publishCompletionUrl' in metadata:`
     ` tr_url = metadata['publishCompletionUrl']`
     `segments = tr_url.rpartition('/')`
     `vid_url = 'https://courses.edx.org{}/transcript/download'.format(segments[0])`

and in `def download_video(self, vid_url, save_as):`
![image_2021-06-28_01-14-48](https://user-images.githubusercontent.com/55428472/123561353-8fb2ae80-d7b0-11eb-980f-90780c25bb37.png)

      def download_video(self, vid_url, save_as):
        # Download the video
        r = requests.get(vid_url)
        srt = r.text
        with open(save_as, 'w', encoding='utf-8') as f:
            for i in srt:
                f.write("{}".format(i))
        return True

**Comment by mmutee** [comment]
I don't know Python either, which script file should I be editing to take advantage of this? I am assuming the whole course needs to be downloaded again in order to get the subtitle files? Have you looked into pull requests, so that your change can be submitted to the project?

**Comment by mhered** [comment]
Hello @asuragan did you manage to download the subtitles with this change?

**Comment by asuragan** [comment]
> Hello @asuragan did you manage to download the subtitles with this change?

yes, but I overwrote the video download code and I have two versions of lib.py, one to download video and one to download subtitles and I don't know how to merge them together

screenshot of downloaded subtitles
![3](https://user-images.githubusercontent.com/55428472/127856553-ff50074b-9406-4019-8919-ada8beeba03a.png)

lib.py for downloading subtitles 
C:\Users\  USER_NAME  \AppData\Local\Programs\Python\Python39\Lib\site-packages\edxdownloader
[lib.zip](https://github.com/rehmatworks/edx-downloader/files/6916570/lib.zip)

**Comment by Kamrulhasan12345** [comment]
@asuragan I have combined everything for downloading both videos and subtitles, and tested that it works

**Comment by rehmatworks** [comment]
A complete rewrite to v2.0 has been published, and that should fix this bug and offer several other improvements.
